scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala

```diff
@@ -237,9 +237,9 @@ class PairSCollectionFunctions[K, V](val self: SCollection[(K, V)])
    */
   def hashJoin[W: ClassTag](that: SCollection[(K, W)])
   : SCollection[(K, (V, W))] = self.transform { in =>
-    val side = that.asMultiMapSideInput
+    val side = that.map(kv => Map(kv._1 -> Seq(kv._2))).sum.asSingletonSideInput
```

Comment:
You can use `asSingletonSideInput(Seq.empty[W])`, and you would need `getOrElse` later on.